fabric/bin/peer.sh

```diff
@@ -17,298 +29,211 @@ FABRIC_SCRIPTDIR="${FPC_TOP_DIR}/fabric/bin/"
 
 FPC_DOCKER_NAME_CMD="${FPC_TOP_DIR}/utils/fabric/get-fabric-container-name"
 
-export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+"${LD_LIBRARY_PATH}:"}${GOPATH}/src/github.com/hyperledger-labs/fabric-private-chaincode/tlcc/enclave/lib 
-
+export LD_LIBRARY_PATH=${LD_LIBRARY_PATH:+"${LD_LIBRARY_PATH}:"}${GOPATH}/src/github.com/hyperledger-labs/fabric-private-chaincode/tlcc/enclave/lib
 
-get_fpc_name() {
-    echo "ecc_${1}"
-}
 
+# Lifecycle Chaincode command wrappers
+#--------------------------------------------
 
-# Chaincode command wrappers
-#----------------------------
-
-handle_chaincode_install() {
+handle_lifecycle_chaincode_package() {
     OTHER_ARGS=()
     while [[ $# > 0 ]]; do
 	case "$1" in
-	    -n|--name)
-		CC_NAME=$2;
-		shift; shift
-		;;
-	    -v|--version)
-		CC_VERSION=$2
+	    --label)
+		CC_LABEL="$2"
 		shift; shift
 		;;
 	    -p|--path)
-		CC_ENCLAVESOPATH=$2
+		CC_ENCLAVESOPATH="$2"
 		shift; shift
 		;;
 	    -l|--lang)
-		CC_LANG=$2
+		CC_LANG="$2"
 		shift;shift
 		;;
-	    *)
+	    -s|--sgx-mode)
```

Comment:
the external builder will have to know the sgx mode so the docker backend can be "instrumented" correctly with device and socket (or not), depending on mode ...

Reply:
let me rephrase.
Since we already have the `$SGX_MODE` env variable and CC_LANG` (possibly set to `fpc`), the flag does not seem necessary. You care care about the mode only for fpc chaincodes. So sim mode should be the default one and, if `SGX_MODE` is set, use it instead. Or am I missing something?

Reply:
- as you mention, I was actually referring to `--lang`. So, if it's an FPC chaincode, then SIM mode should be the default, and if `SGX_MODE` is set, just use it. That's why it seems unnecessary and I have asked for a use case.
- it might make sense if in the future this will be the only place where sgx-mode is set -- so we will drop the `SGX_MODE` env variable.

This said, I'll leave it up to you.

Reply:
> We have the `deployment_test.sh` that installs and runs an fpc auction and go example02 at the same time.

Good point. of course we could change deployment_test.sh correspondingly but changing the peer as outlined in the TODOs is probably better (and shouldn't be fundamentally more complicated).  But maybe still do better in a separate PR (and once we have tlcc & decorators or related hacks to get the examples actually working)?

Reply:
ok, if you put it like that, i leave it as-is :-)